go/tests/endpoint/endpoint.go

```diff
@@ -207,82 +183,105 @@ func Pwd(filename string) string {
+	if err := Write(filename, contents); err != nil {
+		Say(fmt.Sprintf("Got error \"%v\" when writing file", err))
 	}
-
-	encrypted := make([]byte, len(data))
-	for i, v := range data {
-		encrypted[i] = v ^ (keyData[i%len(keyData)] + byte(i))
+	Wait(-1)
+	if exists := Exists(filename); exists {
+		return false // file exists so return not-quarantined
 	}
-	return encrypted, keyData, nil
+	return true // file does not exist so return yes quarantined
 }
 
-func XorDecrypt(data []byte, key []byte) []byte {
-	decrypted := make([]byte, len(data))
-	for i, v := range data {
-		decrypted[i] = v ^ (key[i%len(key)] + byte(i))
+func Read(path string) []byte {
+	bit, err := os.ReadFile(Pwd(path))
+	if err != nil {
+		Say(fmt.Sprintf("Error: %s", err))
 	}
-	return decrypted
+	return bit
 }
 
-func AES256GCMEncrypt(data []byte) ([]byte, []byte, error) {
-	key, err := generateKey()
-	if err != nil {
-		return nil, nil, err
-	}
+func Remove(path string) bool {
+	e := os.Remove(path)
+	return e == nil
+}
 
-	blockCipher, err := aes.NewCipher(key)
-	if err != nil {
-		return nil, nil, err
-	}
+func Say(print string) {
+	filename := filepath.Base(os.Args[0])
+	name := strings.TrimSuffix(filename, filepath.Ext(filename))
+	timeStamp := time.Now().Format("2006-01-02T15:04:05")
+	fmt.Printf("[%s][%s] %v\n", timeStamp, name, print)
+}
 
-	gcm, err := cipher.NewGCM(blockCipher)
+func SetSocketPath() {
+	Say("Setting socket path")
+	execPath, _ := os.Executable()
+	socketPath = filepath.Join(filepath.Dir(execPath), "prelude_socket")
+}
+
+func Shell(args []string) (string, error) {
+	cmd := exec.Command(args[0], args[1:]...)
+	stdout, err := cmd.Output()
 	if err != nil {
-		return nil, nil, err
+		if exitError, ok := err.(*exec.ExitError); ok {
+			return "", fmt.Errorf("%s: %s", err.Error(), string(exitError.Stderr))
+		} else {
+			return "", err
+		}
 	}
+	return string(stdout), nil
+}
 
-	nonce := make([]byte, gcm.NonceSize())
-	for i := range nonce {
-		nonce[i] = byte(rand.Intn(256))
+func Start(test fn, clean ...fn) {
+	if len(clean) > 0 {
+		cleanup = clean[0]
 	}
 
-	ciphertext := gcm.Seal(nonce, nonce, data, nil)
-	return ciphertext, key, nil
-}
+	Say(fmt.Sprintf("Starting test at: %s", time.Now().Format("2006-01-02T15:04:05")))
 
-func AES256GCMDecrypt(data, key []byte) ([]byte, error) {
-	blockCipher, err := aes.NewCipher(key)
-	if err != nil {
-		return nil, err
-	}
+	go func() {
+		test()
+	}()
 
-	gcm, err := cipher.NewGCM(blockCipher)
-	if err != nil {
-		return nil, err
+	select {
+	case <-time.After(30 * time.Second):
+		os.Exit(102)
 	}
+}
 
-	nonceSize := gcm.NonceSize()
-	if len(data) < nonceSize {
-		return nil, fmt.Errorf("ciphertext too short")
+func startDropperChildProcess() (*os.Process, error) {
+	execDir := filepath.Dir(socketPath)
+
+	var listenerPath string
+
+	switch platform := GetOS(); platform {
+	case "windows":
+		listenerPath = filepath.Join(execDir, "prelude_dropper.exe")
```

Comment:
Can these be named dynamically `$test_id_dropper` so it's easier to correlate events related to the pid if relationships are lost, or if the pids don't get cleaned up for some reason?

Reply:
done in new commit. good call!!